lua/paq.lua

```diff
@@ -91,9 +91,15 @@ local function lock_load()
 end
 
 local function state_write()
+   -- remove run key since can have a function in it, and
+    -- json.encode doesn't support functions
+    local pkgs = packages
```

Comment:
This should be a copy, otherwise it'll mutate `packages`.
```suggestion
    local pkgs = vim.deepcopy(packages)
```